File: packages/ios-sdk/Sources/GateFlowKit/Logger/LogLevel.swift
import Foundation

/// Log severity level.
public enum LogLevel: String, CaseIterable {
    case debug = "DEBUG"
    case info = "INFO"
    case warn = "WARN"
    case error = "ERROR"

    public var priority: Int {
        switch self {
        case .debug: 0
        case .info: 1
        case .warn: 2
        case .error: 3
        }
    }
}

File: packages/android-sdk/gateflow/src/main/java/com/gateflow/sdk/logger/LogLevel.kt
package com.gateflow.sdk.logger

import android.util.Log

/** Log severity level. */
enum class LogLevel(val priority: Int) {
    DEBUG(0),
    INFO(1),
    WARN(2),
    ERROR(3);

    internal val logPriority: Int
        get() = when (this) {
            DEBUG -> Log.DEBUG
            INFO -> Log.INFO
            WARN -> Log.WARN
            ERROR -> Log.ERROR
        }
}
